src/Adapter/ArrayAdapter.php

```diff
@@ -72,7 +72,7 @@ public function getData(DataTableState $state): ResultSetInterface
 
         $data = iterator_to_array($this->processData($state, $this->data, $map));
 
-        $length = $state->getLength();
+        $length = $state->getLength() ?? 0;
         $page = $length > 0 ? array_slice($data, $state->getStart(), $state->getLength()) : $data;
```

Comment:
You're ignoring the start value now for unlimited length, seems wrong?

I would expect something like:
```
$page = $array_slice($data, $state->getStart(), $state->getLength() ?? count($data));
```
Which would translate to 'respect $start but max out at length or unlimited if null'.

Reply:
All the other adapters also ignore start if length is not a positive integer, even before I touched them. I assumed this was by design.

Reply:
Hmmm no it seems to be quite inconsistent actually - Mongo does respect the start value currently, ORM doesn't.

I think we should make it consistent one way or another if we're breaking this anyway. Given that I don't see a real use case for respecting `start` with unlimited length I'm fine with ignoring it. Did you accidentally 'fix' this in Mongo?

Reply:
No, I didn't touch whatever was done with the `start` property other than the sanitation + warning I added to its setter method.

Reply:
Modified Mongo to be consistent with the other adapters. If we break we should break consistently.